### DISCUSSION TITLE
How to prioritize sources

### DISCUSSION THREAD

**Discussion by Mtayrn** [spine-tools/SpineOpt.jl#456] [Support]
Hi!

I am working on the optimization of a decentralized energy system that includes PV generation and storage, the objective being to use electricity from those sources before taking from the grid.

But I cannot manage this issue: how do you set the priority when you have several energy sources? 

For example, in your live demo, you had a wind farm and a gas plant. Those plants were configured in the Spine Editor and then the simulation was run. To fulfil the electricity demand, how does the model know what source to use in priority and where can it be configured?

Thank you for your help.

Mats

**Comment by mihlema** [comment]
Hi Mats,

thanks for your question. Maybe what I should mention first that SpineOpt is an optimization tool rather than a simulation tool. 

So in your example, with the wind farm and the gas plant, the user simply provides all the technical details and costs associated with those two generators and SpineOpt will find the dispatch that minimizes the total system costs.

Does that answer your question?

**Reply by Mtayrn** [reply]
That answers my question. Thank you for your reply and clarification!
